[WIP][splash-screen] Extract SplashScreen to unimodule

bbarthec opened the pull request:
# Why

Resolves #6014

# How

## Android
![excalidraw-splashscreen-bare-flow](https://user-images.githubusercontent.com/16623003/73872888-d79a0400-4850-11ea-9155-b45b4d03ce76.png)

# Test Plan
## Android

### Expo Client
- start local `home` project
- launch Android and see SplashScreen is presented

### Managed flow
- from freshly launched ExpoClient select any old snack (SDK33, SDK34, SDK35, SDK36) that contains some SplashScreen/AppLoading logic // TODO @bbarthec: provide one snacks
- test out `app.json` configuration in `/apps/sandbox`

### Bare flow
- navigate to `/apps/bare-expo` and adjust it // TODO @bbarthec: provide adjustments
- launch `yarn expo-splash-screen ...` to test out this cli command and launch project seeing everything is nicely applied

### Standalone app built by turtle
- TODO @bbarthec: has to be tested

# Snapshots from testing

## Android

### bare flow
![splashscreen-bare-flow](https://user-images.githubusercontent.com/16623003/73875245-40837b00-4855-11ea-8282-8edd3098142a.gif)

brentvatne commented on packages/expo-splash-screen/README.md:
```diff
@@ -0,0 +1,37 @@
+# expo-splash-screen
+
+ExpoSplashScreen standalone module
```
```suggestion
Customize the splash screen that is shown on app launch.
```

bbarthec committed:
[splash-screen][Android] Create unimodule for splashscreen

Unimodule features:
 - SplashScreen is mounted under ContentView
 - SplashScreen's image & backgroundColor are configurable
 - SplashScreen would automatically hide when view hierarchy is mounted in located RootView
 - SplashScreen would appear again when RootView has no children again ('reload' behaviour from react-native)

bbarthec committed:
[splash-screen-command][Android] Create cli command for installing splashscreen in bare flow

Features
 - Provide background color & optional png splash image
 - Provide resizeMode & platform
 - Configure native drawable resources for SplashScreen
 - Install SplashScreen.show in MainActivity
 - Preconfigure StatusBar as translucent and transparent during SplashScreen phase

bbarthec committed:
[splash-screen] Remove SplashScreen and AppLoading from expo package

bbarthec committed:
[splash-screen][home] Use splash-screen unimodule

bbarthec committed:
[splash-screen][Android][Client] Use splash-screen unimodule

Changes:
 - Replace LoadingView with SplashScreen view from unimodule and android Popup to indicate loading progress
 - Make HomeActivity use splash-screen unimodule
 - Make old SDKs use splash-screen unimodule (backward compability)
 - Make ExeperienceActivity use splash-screen unimodule
 - Show SplashScreen upon every Experience reloading

bbarthec committed:
[splash-screen] Rename mode to resizeMode

bbarthec committed:
[splash-screen][iOS][bare] Create SplashScreen unimodule for bare flow

bbarthec committed:
[splash-screen][iOS][managed][WIP] Use splash-screen unimodule in ExpoClient

bbarthec committed:
[splash-screen-command][iOS][WIP] Create cli command for installing splashscreen in bare flow

bbarthec committed:
[splash-screen][bare-expo] Demo bare flow app demonstrating expo-splash-screen
